### PR TITLE
[OFFAPPS-1158] Bring user data inline with other apps

### PR DESCRIPTION
**Pull request by ocke**
## Description
User Data App was an earlier app with a lot of experimental code. This big PR tried to make this app more in line with the other apps we have built in the past year.

Every commit has an explanation of what that specific commit tries to achieve. I can take out particular commits when needed.

Go by commit for easier reviewing.

The main take away is that I **removed** the storage functions. It made for unreadable and hard to follow code. It also resulted in bugs, because sometimes the storage had strange values that were coming out of other events.

Second I moved the helpers around getting data, and their processing function into the **api_helpers** file, so it's separated from the app code.

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-1158)

## CCs
@zendesk/apps-migration

## Risks
* [low] User Data App has any issues

### CODE REVIEW

File: src/javascript/app.js [hunk 1]
```diff
@@ -433,15 +432,19 @@ const app = {
   },
 
   onCogClick: function () {
-    const html = renderAdmin({
-      fields: storage('userFields'),
-      orgFields: storage('organizationFields'),
-      orgFieldsActivated: setting('orgFieldsActivated'),
-      hideEmptyFields: setting('hideEmptyFields')
+    return Promise.all([
+      app.getOrganizationFields()
```
Comment (keitheous) [inline src/javascript/app.js]: Considering that there is only one promise, is there any difference in using and not using `Promise.all` here?

File: src/javascript/app.js [hunk 2]
```diff
@@ -317,12 +315,10 @@ const app = {
...
 
   fillEmptyStatuses: function (list) {
-    return reduce(TICKET_STATUSES, function (list, key) {
-      if (!list[key]) {
-        list[key] = '-'
-      }
-      return list
-    }, list)
+    return reduce(TICKET_STATUSES, function (memo, status) {
+      memo[status] = list[status] ? list[status] : '-'
```
Comment (keitheous) [inline src/javascript/app.js]: is this meant to be a single '=' or triple '==='

File: spec/api_helpers.spec.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  })
+
+  describe('#getOrganizationFields', () => {
+    before(() => {
+      sandbox.stub(helpers, 'ajax').callsFake(() => Promise.resolve({ organization_fields: [ { title: 'test', active: true } ] }))
+      sandbox.stub(apiHelpers, 'isOrganizationEditable').callsFake(() => Promise.resolve(true))
+      sandbox.stub(apiHelpers, 'isOrganizationNotesEditable').callsFake(() => Promise.resolve(true))
+    })
+
+    it('it combines the standard fields with the ajax "getOrganizationFields" requests', (done) => {
```
Comment (hoangtrvu) [inline spec/api_helpers.spec.js]: duplicate `it` 🔍

File: spec/app.spec.js [hunk 4]
```diff
@@ -126,185 +87,177 @@ describe('App', () => {
...
-      app.getTickets('requester', 1).then((result) => {
-        assert(ajaxStub.withArgs('searchTickets', 'requester:1').called)
-        assert.deepStrictEqual(result, { open: '1' })
+    it('routes to the requester when clicked on the name', (done) => {
+      document.querySelector('.card.user .contacts .name a').click()
+      setTimeout(() => {
+        assert(invokeSpy.withArgs('routeTo', 'nav_bar', '', sinon.match('requester/requested_tickets')).called)
         done()
-      })
+      }, 5)
```
Comment (hoangtrvu) [inline spec/app.spec.js]: There's a bunch of magic number `5` being used in this file, can we tidy it up a bit? For example, declare a constant once with proper comment (why is this needed, why 5, what if 5 is not always enough, etc).

File: src/javascript/app.js [hunk 5]
```diff
@@ -176,22 +172,20 @@ const app = {
     return eClient.get('ticket.id').then((ticketId) => {
       return ajax('getTicketAudits', ticketId)
     }).then((data) => {
-      each(data.audits, (audit) => {
+      let spokeData
+
+      each(data.audits.audits, (audit) => {
```
[the remark this answers is not in the document]
Comment (ocke) [reply src/javascript/app.js]: I checked it, and you're right. I changed it back

File: src/javascript/lib/ticket_statuses.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export default ['new', 'open', 'pending', 'hold', 'solved', 'closed']
```
Comment (keitheous) [inline src/javascript/lib/ticket_statuses.js]: 👍 no more magic
